optima/plotting.py

```diff
@@ -429,56 +429,63 @@ def plotimprovement(results=None, figsize=(14,10), lw=2, titlesize=14, labelsize
 ##################################################################
     
     
-def plotallocs(results=None, figsize=(14,10), **kwargs):
-    ''' 
-    Plot multiple allocations on bar charts -- intended for scenarios and optimizations.
-    Results object must be of Multiresultset type.
-    
-    Version: 2016jan24    
-    '''
+def plotallocs(multires=None, which=None, compare=False):
+    ''' Plot multiple allocations on bar charts -- intended for scenarios and optimizations '''
     
-    # Validate input
-    if not(hasattr(results, 'budget')): raise OptimaException('No budget found for results object:\n"%s"' % results)
+    if which is None: which = ['budget','coverage'] # Plot everything if not specified
+    elif type(which) in [str, tuple]: which = [which] # If single value, put inside list
+
+    # Preliminaries: process inputs and extract needed data
+    if 'budget' in which: budgetstoplot = [budget for budget in multires.budget.values() if budget]
+    if 'coverage' in which: coveragetoplot = [coverage for coverage in multires.coverage.values() if coverage]
+    budgetyearstoplot = [budgetyears for budgetyears in multires.budgetyears.values() if budgetyears]
     
-    # Preliminaries: extract needed data
-    budgetstoplot = [budget for budget in results.budget.values() if budget]
-    budgetyearstoplot = [budgetyears for budgetyears in results.budgetyears.values() if budgetyears]
-    proglabels = budgetstoplot[0].keys() 
-    alloclabels = [key for k,key in enumerate(results.budget.keys()) if results.budget.values()[k]] # WARNING, STUPENDOUSLY UGLY
-    nprogs = len(proglabels)
-    nallocs = len(alloclabels)
+    for plotkey in which:
+        toplot = budgetstoplot if plotkey=='budget' else coveragetoplot
+        proglabels = toplot[0].keys() 
+
+        alloclabels = [key for k,key in enumerate(multires.__dict__[plotkey].keys()) if multires.__dict__[plotkey].values()[k]] 
```

Comment:
i generally prefer
`getattr(multires, plotkey)`
to
`multires.__dict__[plotkey]`
since shorter and slightly more transparent...